packages/truffle-core/lib/build.js

```diff
@@ -74,7 +74,7 @@ const Build = {
     // Duplicate build directory for legacy purposes
     options.destination_directory = options.build_directory;
 
-    if (typeof builder === "undefined") {
+    if (builder === null) {
```

Comment:
Should we also allow `undefined`? Just in case? (ooh, a new github feature)
```suggestion
    if (builder === null || typeof builder === "undefined") {
```